src/skadi/util.cc

```diff
@@ -71,7 +76,8 @@ weighted_grade(const std::vector<double>& heights,
   // Get the average weighted grade by homogenizing total weight. Return
   // max grades (up and down) and mean elevation.
   return std::make_tuple(total_grade * (1.0 / total_weight), max_up_grade, max_down_grade,
-                         (total_elev / n));
+                         (has_only_no_data ? static_cast<double>(baldr::kNoElevationDataRaw)
+                                           : total_elev / n));
```

Comment:
we can't move all of the logic since the caller has no chance of knowing whether the heights were all on sea level or no_data values. actually the latter would've caused a "division by zero" error so far. guess that never happened until now